### PR TITLE
Feature/refactor universe

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/LedgerRecoveryModule.java [hunk 1]
```diff
@@ -43,17 +58,54 @@
...
+		RadixEngine<LedgerAtom> radixEngine,
+		CommittedAtomsStore store,
+		@Genesis Atom atom,
+		Hasher hasher,
+		Serialization serialization,
+		ValidatorSetBuilder validatorSetBuilder
+	) throws RadixEngineException {
+		final ClientAtom genesisAtom = ClientAtom.convertFromApiAtom(atom, hasher);
+		if (!store.containsAID(genesisAtom.getAID())) {
+			RadixEngineBranch<LedgerAtom> branch = radixEngine.transientBranch();
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/LedgerRecoveryModule.java:74]: I think it's better to move this code into dedicated method.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/LedgerRecoveryModule.java [hunk 2]
```diff
@@ -85,33 +134,34 @@ private VerifiedVertexStoreState vertexStoreState(
 		Hasher hasher
 	) {
 		return ledgerEntryStore.loadLastVertexStoreState()
-				.map(serializedVertexStoreState -> {
-					UnverifiedVertex root = serializedVertexStoreState.getRoot();
-					HashCode rootVertexId = hasher.hash(root);
-					VerifiedVertex verifiedRoot = new VerifiedVertex(root, rootVertexId);
+			.filter(vertexStoreState -> vertexStoreState.getHighQC().highestQC().getEpoch() == lastEpochProof.getEpoch() + 1)
+			.map(serializedVertexStoreState -> {
+				UnverifiedVertex root = serializedVertexStoreState.getRoot();
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/LedgerRecoveryModule.java:139]: I think it's better to move this code into dedicated method.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/LedgerRecoveryModule.java [hunk 3]
```diff
@@ -85,33 +134,34 @@ private VerifiedVertexStoreState vertexStoreState(
...
-				});
+				return VerifiedVertexStoreState.create(
+					serializedVertexStoreState.getHighQC(),
+					verifiedRoot,
+					vertices,
+					serializedVertexStoreState.getHighestTC()
+				);
+			})
+			.orElseGet(() -> {
+				UnverifiedVertex genesisVertex = UnverifiedVertex.createGenesis(lastEpochProof.getRaw());
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/LedgerRecoveryModule.java:155]: I think it's better to move this code into dedicated method.
Also, wider adoption of 'var' will reduce the number of repetitions of type names.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/checkpoint/CheckpointUtils.java [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+		);
+
+		List<ParticleGroup> particleGroups = new ArrayList<>();
+		for (final var entry : stakesByKey.entrySet()) {
+			final var particles = Lists.newArrayList(tokensByKey.get(entry.getKey()));
+			final var delegationsOfKey = entry.getValue();
+
+			final var stakerAddress = new RadixAddress(magic, entry.getKey());
+
+			ParticleGroup.ParticleGroupBuilder builder = ParticleGroup.builder();
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/checkpoint/CheckpointUtils.java:201]: Whole particle group building code deserves dedicated method.

**Review by siy — APPROVED** [review]
LGTM with comments.

**Review by LukasGasior1 — APPROVED** [review]

**Pull request merged**